### PR TITLE
feat: Not found page

### LINKED ISSUE #183: Error page (404 and 500 errors)
## Description
Create error page templates/pages for 404 and 500 errors.
Non Engr tasks: https://trello.com/c/efmFDKjR/40-list-global-error-state-and-draft-helpful-error-messages

MVP:
- [x]  simple 404 error page design
- [ ] implementation

### PR DESCRIPTION
**Pull request by josebui**
## Description

- Added not found page

### Checklist
- [x] Corresponding issue has been opened
- [ ] New tests added
- [x] Strings are localized
- [x] Verified on mobile
- [x] Verified on desktop
- [x] Verified accessibility ([Accessibility - Web applications](https://docs.google.com/document/d/1VwPDyLqw7r_iISMgDZr1FdAmAqqOIIVIxQqsYA-HaHE/edit?usp=sharing))

### Related Issues
Fixes #183 

### Verification steps

Screenshots:

download (47)
download (46)

### CODE REVIEW

File: src/layout/NotFound.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+              marginTop: 1,
+              fontSize: '16px',
+              textTransform: 'uppercase',
+            })}
+          >
+            {t('common.not_found_error_code')}
+          </Typography>
+          <Typography sx={{ marginTop: 4, fontSize: '16px' }}>
+            <Trans i18nKey="common.not_found_help_message">
+              Prefix <Link href={HELP_URL}>link</Link>.
```
Comment (cacarrara) [inline src/layout/NotFound.js:40]: Doubt: when this `Prefix ...` message will be used?

**Review by cacarrara — APPROVED** [review]
Overall it LGTM :+1:

File: src/navigation/Navigation.js [hunk 2]
```diff
@@ -13,15 +13,15 @@ const PAGES = {
...
   },
   '/groups': {
     label: 'navigation.groups',
-    match: path => path.startsWith('/groups'),
+    match: path => path === '/groups' || path.startsWith('/groups/'),
   },
   '/tools': {
     label: 'navigation.tools',
-    match: path => path.startsWith('/tools'),
+    match: path => path === '/tools' || path.startsWith('/tools/'),
```
Comment (paulschreiber) [inline src/navigation/Navigation.js:24]: question: why doesn't `startsWith('/tools'`)` do what we want?

**Review by paulschreiber — CHANGES_REQUESTED** [review]
issue: let's move the body context so it is left-aligned, like the other pages are.

image

File: src/layout/NotFound.js (hunk 1 above)
Comment (josebui) [reply src/layout/NotFound.js:40]: @cacarrara It is just to identify the parts for the translation, it is used by i18next to identify the parts that will be replaced on each locale

File: src/navigation/Navigation.js (hunk 2 above)
Comment (josebui) [reply src/navigation/Navigation.js:24]: @paulschreiber `startsWith('/tools')` picks anything that starts with `/tools` but that is not exaclty what we need. The user is under or in `tools` if the route is exactly `/tools` or under `/tools/` but for example a different thing that start with `tools` like `tools123` should not be identified as `tools`, does that make sense?

File: src/layout/NotFound.js (hunk 1 above)
Comment (josebui) [reply src/layout/NotFound.js:40]: This one has 3 parts `prefix` `link` `period`

File: src/navigation/Navigation.js (hunk 2 above)
Comment (paulschreiber) [reply src/navigation/Navigation.js:24]: What about something like`startsWith('/tools(/.+)?')`?

Comment (josebui) [reply src/navigation/Navigation.js:24]: sure, a regular expression might be able to handle this too, I will test that

**Review by paulschreiber — APPROVED** [review]

**Pull request merged**